src/unit_tests/shared/test_atomic.c

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2015-2021, Wazuh Inc.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <cmocka.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <string.h>
+
+#include "../../headers/atomic.h"
+#include "../wrappers/posix/pthread_wrappers.h"
+
+atomic_int_t test_variable = ATOMIC_INT_INITIALIZER(0);
+
+
+static int setup_variable(void **state) {
+    expect_function_call(__wrap_pthread_mutex_lock);
+    expect_function_call(__wrap_pthread_mutex_unlock);
+    atomic_int_set(&test_variable, 1231);
+    return 0;
+}
+
+static int teardown_variable(void **state) {
+    expect_function_call(__wrap_pthread_mutex_lock);
+    expect_function_call(__wrap_pthread_mutex_unlock);
+    atomic_int_set(&test_variable, 0);
+    return 0;
+}
```

Comment:
These setup and teardown functions don't make much sense, they look the same as a test with no asserts.

I thinks a better approach would be to directly create the variable under test locally to each test and initialize it with the desired value directly.